integration-tests/tests/src/tests/mixed.ts

```diff
@@ -0,0 +1,313 @@
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm from "realm";
+import { Decimal128, ObjectId, UUID } from "bson";
+import { expect } from "chai";
+import { openRealmBefore } from "../hooks";
+
+type ISingleSchema = {
+  a: Realm.Mixed;
+  b: Realm.Mixed;
+  c: Realm.Mixed;
+  d: Realm.Mixed;
+};
+
+type IVertexSchema = {
+  a: number;
+  b: number;
+  c: number;
+};
+
+type IMixNestedSchema = {
+  a: Realm.Mixed;
+  b: Realm.Mixed;
+  c: Realm.Mixed;
+  d: Realm.Mixed;
+};
+
+type IMixedNullableSchema = {
+  nullable: Realm.Mixed;
+  nullable_list: Realm.Mixed[];
+};
+
+type IMixedSchema = {
+  value: Realm.Mixed;
+};
+
+const SingleSchema: Realm.ObjectSchema = {
+  name: "mixed",
+  properties: {
+    a: "mixed",
+    b: "mixed",
+    c: "mixed",
+    d: "mixed",
+  },
+};
+
+const VertexSchema: Realm.ObjectSchema = {
+  name: "Vertex",
+  properties: {
+    a: "int",
+    b: "int",
+    c: "int",
+  },
+};
+
+const MixNestedSchema: Realm.ObjectSchema = {
+  name: "Nested",
+  properties: {
+    a: "mixed",
+    b: "mixed",
+    c: "mixed",
+  },
+};
+
+const MixedNullableSchema: Realm.ObjectSchema = {
+  name: "mixed",
+  properties: {
+    nullable: "mixed",
+    nullable_list: "mixed[]",
+  },
+};
+
+const MixedSchema: Realm.ObjectSchema = {
+  name: "MixedClass",
+  properties: { value: "mixed" },
+};
+
+describe("Mixed", () => {
+  describe("primitive values", () => {
+    openRealmBefore({ schema: [SingleSchema] });
+    it("support primitives", function (this: RealmContext) {
+      this.realm.write(() => this.realm.create(SingleSchema.name, { a: "xxxxxx", b: 555, c: true }));
+      const data = this.realm.objects<ISingleSchema>(SingleSchema.name)[0];
+      expect(data.a).equals("xxxxxx", "should store xxxxxx");
+      expect(data.b).equals(555, "should store 555");
+      expect(data.c).equals(true, "should store boolean (true)");
+    });
+
+    it("support queries on mixed primitives", function (this: RealmContext) {
+      const N = 100;
+      for (let i = 0; i < N; i++) {
+        this.realm.write(() => this.realm.create(SingleSchema.name, { a: "xxxxxx", b: "xxx", c: i }));
+      }
+
+      const data = this.realm.objects(SingleSchema.name);
+      const section = data.filtered('a BEGINSWITH "x" AND b BEGINSWITH "x"');
+      const half = data.filtered("c < 50");
+      expect(section.length).equals(N, "We expect only 100 items.");
+      expect(half.length).equals(N / 2, "We expect only 50 items.");
+    });
+
+    it("support complex primitive types", function (this: RealmContext) {
+      const d128 = Decimal128.fromString("6.022e23");
+      const oid = new ObjectId();
+      const uuid = new UUID();
+      const date = new Date();
+
+      const data: ISingleSchema = this.realm.write(() =>
+        this.realm.create(SingleSchema.name, { a: oid, b: uuid, c: d128, d: date }),
+      );
```

Comment:
You might as well move the type to the `create` so you also get correct types on the second "values" argument too.

```suggestion
      const data = this.realm.write(() =>
        this.realm.create<ISingleSchema>(SingleSchema.name, { a: oid, b: uuid, c: d128, d: date }),
      );
```